Modules/Loadable/Segmentations/Logic/vtkSegmentEditorLogic.cxx

```diff
@@ -0,0 +1,2182 @@
+    }
+    if (defaultTerminologyEntryStr.empty())
+    {
+      defaultTerminologyEntryStr = DefaultTerminologyEntry;
+    }
+
+    if (!defaultTerminologyEntryStr.empty())
+    {
+      addedSegment->SetTerminology(defaultTerminologyEntryStr);
+    }
+  }
+
+  // Set segment status to one that is visible by current filtering criteria
+  vtkSlicerSegmentationsModuleLogic::SetSegmentStatus(addedSegment, segmentStatus);
+
+  // Select the new segment
+  if (!addedSegmentID.empty())
+  {
+    SegmentEditorNode->SetSelectedSegmentID(addedSegmentID.c_str());
+  }
+
+  // Assign the new segment the terminology of the (now second) last segment
+  if (segmentationNode->GetSegmentation()->GetNumberOfSegments() > 1)
+  {
+    vtkSegment* secondLastSegment = segmentationNode->GetSegmentation()->GetNthSegment(segmentationNode->GetSegmentation()->GetNumberOfSegments() - 2);
+    std::string repeatedTerminologyEntry = secondLastSegment->GetTerminology();
+    segmentationNode->GetSegmentation()->GetSegment(addedSegmentID)->SetTerminology(repeatedTerminologyEntry);
+  }
+  return addedSegmentID;
+}
+
+//------------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanAddSegments() const
+{
+  // Disable adding new segments until source volume is set (or reference geometry is specified for the segmentation).
+  // This forces the user to select a source volume before start adding segments.
+  return (GetSegmentation() != nullptr) && (GetSourceVolumeNode() != nullptr) && (!GetReferenceImageGeometryString().empty());
+}
+
+//-----------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanRedo() const
+{
+  return SegmentationHistory && SegmentationHistory->IsRestoreNextStateAvailable();
+}
+
+//------------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanRemoveSegments() const
+{
+  // Only enable remove button if a segment is selected
+  return IsSegmentIdValid(GetCurrentSegmentID());
+}
+
+//-----------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanTriviallyConvertSourceRepresentationToBinaryLabelMap() const
+{
+  if (!IsSegmentationNodeValid())
+  {
+    return false;
+  }
+
+  if (GetSegmentation()->GetNumberOfSegments() < 1)
+  {
+    return true;
+  }
+
+  if (GetSegmentation()->GetSourceRepresentationName() == vtkSegmentationConverter::GetSegmentationBinaryLabelmapRepresentationName())
+  {
+    return true;
+  }
+  return false;
+}
+
+//-----------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanUndo() const
+{
+  return SegmentationHistory && SegmentationHistory->IsRestorePreviousStateAvailable();
+}
+
+//-----------------------------------------------------------------------------
+void vtkSegmentEditorLogic::ClearUndoState() const
+{
+  if (SegmentationHistory)
```

Comment:
Re-basing the topic and running `uvx pre-commit run -a` will now include the braces.